src/analysis/retail/mage/frost/core/WintersChillEvent.tsx

```diff
@@ -0,0 +1,157 @@
+    if (!this.hasTwoShatteredSpells() && !this.hasOneSpellAndRayOfFrost()) {
+      message = 'Stacks not used';
+    } else if (this.hasTwoShatteredSpells()) {
+      message = '2 spells shattered';
+    } else if (this.hasOneSpellAndRayOfFrost()) {
+      message = '1 spell and Ray of Frost shattered';
+    }
+    return message;
+  }
+
+  getPrecastPerformance(): QualitativePerformance {
+    let performance = QualitativePerformance.Fail;
+    if (this.precast && this.precastInDamageEvents()) {
+      performance = QualitativePerformance.Good;
+    } else if (!this.precast && this.precastIcicles === 4) {
+      performance = QualitativePerformance.Good;
+    } else if (this.precast && !this.precastInDamageEvents()) {
+      performance = QualitativePerformance.Fail;
+    } else if (!this.precast) {
+      performance = QualitativePerformance.Fail;
+    }
+    return performance;
+  }
+
+  getPrecastPerformanceMessage(): string {
+    let message = 'Unknown fail';
+    if (this.precast && this.precastInDamageEvents()) {
+      message = 'Precast shattered';
+    } else if (!this.precast && this.precastIcicles === 4) {
+      message = 'No precast on 4 icicles';
+    } else if (this.precast && !this.precastInDamageEvents()) {
+      message = 'Precast not shattered';
+    } else if (!this.precast) {
+      message = 'No precast';
+    }
+    return message;
+  }
+
+  getPerformance(): QualitativePerformance {
+    let performance = QualitativePerformance.Fail;
+    if (
+      this.getPrecastPerformance() === QualitativePerformance.Good &&
+      this.getShatterPerformance() === QualitativePerformance.Good
+    ) {
+      performance = QualitativePerformance.Good;
+    }
+    return performance;
+  }
+
+  precastInDamageEvents() {
+    const preCastDamage: DamageEvent | undefined =
+      this.precast && GetRelatedEvent(this.precast, 'SpellDamage');
+    return preCastDamage && this.damageEvents.includes(preCastDamage);
+  }
+
+  getPerformanceDetails(): JSX.Element {
+    const spellsArray: Spell[] = Object.values(SPELLS);
+    const precast: Spell | undefined = spellsArray.find(
+      (spell) => spell.id === this.precast?.ability.guid,
+    );
+    const flurry: Spell | undefined = spellsArray.find(
+      (spell) => spell.id === this.flurry?.ability.guid,
+    );
+    const spells: Spell[] = [];
+    if (precast) {
+      spells.push(precast);
+    }
+    if (flurry) {
+      spells.push(flurry);
+    }
+    this.damageEvents.forEach((damage) => {
+      const spell: Spell | undefined = spellsArray.find(
+        (spell) => spell.id === damage.ability.guid,
+      );
+      if (spell) {
+        spells.push(spell);
+      }
+    });
+    const tooltip = (
+      <>
+        <PerformanceMark perf={this.getPrecastPerformance()} />
+        {this.getPrecastPerformanceMessage()} <br />
+        <PerformanceMark perf={this.getShatterPerformance()} />
+        {this.getShatterPerformanceMessage()} <br />
+        <SpellSeq spells={spells} />
```

Comment:
ideally use block elements instead of `<br/>` for layout. also missing spaces between the checkmarks and the text.

(prettier may not like this suggestion as-is)

```suggestion
        <div><PerformanceMark perf={this.getPrecastPerformance()} />{' '}
        {this.getPrecastPerformanceMessage()}</div>
        <div><PerformanceMark perf={this.getShatterPerformance()} />{' '}
        {this.getShatterPerformanceMessage()} </div>
        <SpellSeq spells={spells} />
```